sycl/source/handler.cpp

```diff
@@ -494,7 +494,7 @@ event handler::finalize() {
       // uint32_t StreamID, uint64_t InstanceID, xpti_td* TraceEvent,
       int32_t StreamID = xptiRegisterStream(detail::SYCL_STREAM_NAME);
       auto [CmdTraceEvent, InstanceID] = emitKernelInstrumentationData(
-          StreamID, MKernel, MCodeLoc, impl->MIsTopCodeLoc, MKernelName.c_str(),
+          StreamID, MKernel, MCodeLoc, impl->MIsTopCodeLoc, MKernelName.data(),
```

Comment:
The `emitKernelInstrumentationData` should be changed to use `std::string_view` as a type of the `SyclKernelName` parameter. Right now, it is `const std::string &SyclKernelName`. So it will cause temporary string object creation when you pass `MKernelName.data()`.

@sergey-semenov Do you mind if I move the changes around this functionality into a separate PR?
It would decrease the amount of changes in this PR and simplify review?

Reply:
Good catch, this draft had a few unrelated changes mixed in.

> @sergey-semenov Do you mind if I move the changes around this functionality into a separate PR?
It would decrease the amount of changes in this PR and simplify review?

That would be helpful.